### PR TITLE
Ubuntu/devel: release 23.1

### PR DESCRIPTION
**Pull request by aciba90**
## Tested:

build-package
sbuild in lunar -> no lintian warnings
installation successful

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,11 @@
+cloud-init (23.1-0ubuntu1) lunar; urgency=medium
+
+  * Upstream snapshot based on upstream/main at 242f6bc4.
+    - Bugs fixed in this snapshot: (LP: #1990513, #1999164, #1931174)
+      (LP: #1815990)
```
Comment (blackboxsw) [inline debian/changelog:5]: Minor diff in generated d/changelog content:
```suggestion
  * Upstream snapshot based on 23.1
    List of changes from upstream can be found at
    https://raw.githubusercontent.com/canonical/cloud-init/23.1/ChangeLog
    - Bugs fixed in this snapshot: (LP: #1990513, #1999164, #1931174)
      (LP: #1815990)
```

**Review by blackboxsw — APPROVED** [review]
Confirmed: 
 - [x] content correct for new_upstream_snapshot.py -c 23.1 once d/changelog updated
 - [x] version good for lunar release based on 23.1 tag: `(23.1-0ubuntu1) lunar`
 - [x] all fixed bugs accounted for in d/changelog
 - [x] build-package/sbuild success

**Comment by blackboxsw** [comment]
Package [23.1 uploaded to lunar](https://launchpad.net/ubuntu/lunar/+queue?queue_state=3&queue_text=cloud-init) We can now tag and push this branch to upstream/ubuntu/devel

**Comment by aciba90** [comment]
Branch and tag pushed. Thanks, @blackboxsw, for the changelog fix and review.

**Pull request merged**